### ISSUE TITLE
Clarify UI for on-demand tasks by way of "Install" button

### ISSUE THREAD

**Issue by binkleybloom** [munki/munki#522]
The sequence "Install" -> "Installing" -> "Install" is confusing for on-demand tasks. Changing the contents of the button specifically for on-demand tasks within Managed Software Center may better communicate the task behavior to the end user.

One [English] idea - 
Initial button: "Run"
During execution: "Processing"
Upon completion: "Completed" - which will switch back to "Run" once MSC has quit and relaunched.

**Comment by gregneagle** [comment]
I agree that the current labels are not ideal.

Your suggestion is good, but doesn't really go far enough.

Not everything that can be installed OnDemand is a script or task; some might actually still be software installs. If you wanted, for example, to enable users to install a new/clean copy of a VM image, "Install" -> "Installing" would absolutely be appropriate.

So what is _really_ needed is a way for the admin to specify labeling on a per-item basis. Not sure if that should be a set of pre-defined labels, or if admins should be able to invent their own. The first approach presents localization issues: a new version of MSC would need to have new labels in all supported languages.

As for the status changing to "Completed" until MSC is re-launched; I'd have to look closely at the code, but on first thought I think that would actually be more difficult than you might think, as it would require accurately storing and using state information for an arbitrary number of items through a managedsoftwareupdate run cycle, which is not tightly tied to MSC.

To summarize: I agree the current labelling is not ideal. The "correct" fix is far from clear and definitely not trivial. It's "on the list", but no ETA at present.

**Comment by binkleybloom** [comment]
This has clearly been on your radar - many valid points I hadn't considered. 

And with that being said, I would happily cast a vote for your second approach and customize these in pkginfo. 

Thanks Greg - appreciate the response.
